vignettes/pkgnet-intro.Rmd

```diff
@@ -0,0 +1,249 @@
+mostRef[,list(node, numDescendants, coverageRatio, totalLines)]
+
+```
+
+|node                   | numDescendants| coverageRatio| totalLines|
+|:----------------------|--------------:|-------------:|----------:|
+|dmicroseconds          |             39|     0.0000000|          1|
+|days_in_months_so_far  |             22|     0.0000000|         10|
+|check_period           |             15|     0.0000000|         12|
+|as.difftime            |             13|            NA|         NA|
+|fast_strptime          |             13|     0.8571429|          7|
+|comp_posix_date        |             12|     0.7692308|         13|
+|date_to_posix          |             12|     1.0000000|          2|
+|ceiling_date           |             10|     0.9666667|         30|
+|add_number_to_duration |              8|     1.0000000|          1|
+|check_interval         |              8|     0.0000000|         11|
+
+Inspecting results such as these can help an R package developer decide which function to cover with unit tests next.  In this case, the most referenced function, `dmicroseconds`, appears to be a simple one line function that may not be necessary to cover in unit testing.  However, the second most referenced function, `days_in_months_so_far`, might benefit from some unit test coverage.
+
+## Discovering Similar Functions
+Looking at that same large package, let's say we wanted to explore options for consolidating functions.  In that case, we could use the `igraph` object to highlight functions with the same dependencies via Jaccard similarity. 
+
+```{r fakeDetail2, eval=FALSE}
+# Get igraph object
+funcGraph <- reportObjWithCovr$PackageFunctionReporter$pkg_graph
+funcNames <- igraph::vertex_attr(funcGraph, name = "name")
+
+# Jaccard Similarity
+sim <- igraph::similarity(graph = funcGraph
+                          , mode = "in"
+                          , method = "jaccard")
+diag(sim) <- 0
+sim[sim < 1] <- 0
+
+simGraph <- igraph::graph_from_adjacency_matrix(adjmatrix = sim, mode = "undirected")
+
+# Find groups with same dependencies (similarity == 1)
+sameDeps <- igraph::max_cliques(graph = simGraph
+                                , min = 2
+                                )
+
+# Write results
+for (i in seq_along(sameDeps)) {
+         cat(paste0("Group ", i, ": "))
+         cat(paste(funcNames[as.numeric(sameDeps[[i]])], collapse = ", "))
+         cat("\n")
+}
+```
+  
+```{r resultFromFake, echo=FALSE, results='markup'}
+cat("Group 1: stamp_time, stamp_date
+Group 2: ms, hm
+Group 3: new_interval, %--%, int_diff
+Group 4: floor_date, quarter, semester
+Group 5: picoseconds, microseconds, nanoseconds, milliseconds
+Group 6: weeks, days, years, seconds_to_period, seconds, new_period, minutes, hours
+Group 7: yq, dmy, ymd_hms, ymd_hm, ymd_h, ymd, ydm_hms, ydm_hm, ydm_h, ydm, pretty_dates, parse_date_time2, parse_date_time, myd, mdy_hms, mdy_hm, mdy_h, mdy, local_time, fast_strptime, dym, dmy_hms, dmy_hm, dmy_h
+"
+)
+```
+
```

Comment:
I was surprised that this ended after the printout stuff. Can you please add two more things?

1) a short interpretation of those groupings above. We need to tell people who don't know what Jaccard Similarity is how to interpret these results, so they'll care enough to go Google it

2) A closing paragraph that starts with `In this vignette, we...` and briefly summarizes what you talked about. I also recommend that the final line be like "If you noticed any errors or ambiguities in this vignette, please [submit a pull request](https://github.com/UptakeOpenSource/pkgnet/blob/master/CONTRIBUTING.md#prs)"